src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+import TodoFilter from '../TodoFilter';
+
+import { deleteTodo } from '../../api/todos';
+import { useTodos } from '../../hooks/useTodos';
+import { Todo } from '../../types';
+
+const Footer: React.FC = () => {
+  // eslint-disable-next-line prettier/prettier
+  const {
+    todos,
+    setTodos,
+    filter,
+    setFilter,
+    setIsAllDeleted,
+    handleError
+  } = useTodos();
+
+  const isClearButtonVisible = todos.some(todo => todo.completed);
+  const amountItemsLeft = todos.filter(todo => !todo.completed).length;
+
+  const handleDeletetodo = (todoId: number) => {
+    setIsAllDeleted(true);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos((prevTodos: Todo[]) =>
+          prevTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        handleError('Unable to delete a todo');
+      })
+      .finally(() => {
+        setIsAllDeleted(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleDeletetodo(todo.id);
+      }
+    });
```

Comment:
Ah, I see. You can create a state `loadingTodoIds` or something similar, and add ids to this array when deleting/toggling them and depending on this state show loaders for the corresponding todos